### PR TITLE
Feat/ice powerup

### PR DESCRIPTION
**Pull request by aaronw84**
Ice item pickup causing Chill PowerUp and FireStone item pickup causing Shocked PowerUp.

### CODE REVIEW

File: TKGame/PowerUps/Components/FireStone PowerUps/C_FireStone_PrimaryAttack.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        private readonly float SHOCKED_TICK_INTERVAL = 0.5f;
+        private readonly float SHOCKED_DAMAGE_PER_TICK = 0f;
+        #endregion Constants
+        ComponentType IComponent.Type => ComponentType.AttackPrimary;
+        public string NameID { get; private set; }
+        public Rectangle HitBox { get; set; }
+        public AttackType AttackType { get; }
+        public bool isAttacking { get; private set; }
+
+        public C_FireStone_PrimaryAttack()
```
Comment (NateCastleOIT) [inline TKGame/PowerUps/Components/FireStone PowerUps/C_FireStone_PrimaryAttack.cs:29]: Is this supposed to still be named this?

Comment (NateCastleOIT) [reply TKGame/PowerUps/Components/FireStone PowerUps/C_FireStone_PrimaryAttack.cs:29]: I see shock stuff in the fire class so not certain why

File: TKGame/PowerUps/Components/FireStone PowerUps/Related Entities/Shocked.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+using System.Threading.Tasks;
+using TKGame.BackEnd;
+using TKGame.Components.Interface;
+using TKGame.Content.Weapons;
+using TKGame.Players;
+using TKGame.Status_Effects;
+
+namespace TKGame.PowerUps.RelatedEntities
+{
+    public class Shocked : Entity
```
Comment (NateCastleOIT) [inline TKGame/PowerUps/Components/FireStone PowerUps/Related Entities/Shocked.cs:16]: This looks good but the name "Shocked" for the entity something spawns is a bit confusing with a "Shocked" status as well

File: TKGame/PowerUps/Components/FireStone PowerUps/Related Entities/Shocked.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            {
+                this.IsExpired = true;
+                return;
+            }
+
+            List<Entity> entities = EntityManager.GetEntities();
+
+            foreach (Entity e in entities)
+            {
+                if (e != this && HitBox.Intersects(e.HitBox) && e != Player.Instance)
```
Comment (NateCastleOIT) [inline TKGame/PowerUps/Components/FireStone PowerUps/Related Entities/Shocked.cs:68]: Whenever I read these loops with hitbox detection I get confused if I read it wrong, we might want to use this. explicitly more often for situations like this

File: TKGame/PowerUps/Components/FireStone PowerUps/Related Entities/Shocked.cs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            {
+                offset = offset * -1 - HitBox.Width;
+            }
+            HitBox = new Rectangle(Player.Instance.HitBox.Center.X + offset, Player.Instance.HitBox.Y, HitBox.Width, HitBox.Height);
+        }
+
+        public override void Draw(SpriteBatch spriteBatch)
+        {
+            base.Draw(spriteBatch);
+        }
```
Comment (NateCastleOIT) [inline TKGame/PowerUps/Components/FireStone PowerUps/Related Entities/Shocked.cs:96]: noice

File: TKGame/PowerUps/Components/Ice PowerUps/C_Ice_PrimaryAttack.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            NameID = "IcePrimaryAttack";
+            AttackType = AttackType.Primary;
+            // does player hitbox always move when attacking? If so, this isn't necessary
+            HitBox = new Rectangle(new Point(Player.Instance.HitBox.X - 150, Player.Instance.HitBox.Y), new Point(150, 150));
+        }
+        public void Update(Entity entity)
+        {
+            // TODO: put in input component
+
+            if (Input.WasKeyPressed(Keys.I))
```
Comment (NateCastleOIT) [inline TKGame/PowerUps/Components/Ice PowerUps/C_Ice_PrimaryAttack.cs:40]: I?

File: TKGame/PowerUps/Components/Ice PowerUps/C_Ice_PrimaryAttack.cs [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            {
+                isAttacking = true;
+                Entity chilled = new Chilled(0.5f, CHILLED_DURATION, CHILLED_TICK_INTERVAL, CHILLED_DAMAGE_PER_TICK, new C_Chilled_Physics(), new C_Chilled_Graphics());
```
Comment (NateCastleOIT) [inline TKGame/PowerUps/Components/Ice PowerUps/C_Ice_PrimaryAttack.cs:43]: Again about the entity that deals the dmg/status effect being the same or similar name to the effect it gives is a bit confusing

**Review by NateCastleOIT — APPROVED** [review]
looks good, just need to finish the TODOs

File: TKGame/PowerUps/Components/FireStone PowerUps/Related Entities/Shocked.cs (hunk 3 above)
Comment (NathanGr33n) [reply TKGame/PowerUps/Components/FireStone PowerUps/Related Entities/Shocked.cs:68]: I've used hitbox detection in loops also, we may just want to add comments to these lines to specify what we are doing exactly

**Review by NathanGr33n — COMMENTED** [review]
Looks good, you may want to specifically comment what you are doing in certain parts to help explain it better, but looks good otherwise.

File: TKGame/PowerUps/Components/FireStone PowerUps/C_FireStone_PrimaryAttack.cs (hunk 1 above)
Comment (aaronw84) [reply TKGame/PowerUps/Components/FireStone PowerUps/C_FireStone_PrimaryAttack.cs:29]: The fire effects were added to the potion item, so I just decided to make the firestone have the shocked effect.

File: TKGame/PowerUps/Components/Ice PowerUps/C_Ice_PrimaryAttack.cs (hunk 5 above)
Comment (aaronw84) [reply TKGame/PowerUps/Components/Ice PowerUps/C_Ice_PrimaryAttack.cs:40]: Just thought of a random key that isn't being used

**Review by keeganhauser — APPROVED** [review]
As Nate mentioned, looks good minus the TODOs, but what's here is good 👍

**Pull request merged**